client/src/components/session-timer/session-timer-mixin.ts

```diff
@@ -0,0 +1,42 @@
+import Vue from 'vue'
+import { Component } from 'vue-property-decorator'
+
+import timerModule from '@/modules/vx-timer'
+import {
+  EXISTING_NR_TIMEOUT_MS,
+  EXISTING_NR_TIMER_NAME,
+  NR_COMPLETION_TIMEOUT_MS,
+  NR_COMPLETION_TIMER_NAME
+} from '@/store/new-request-module'
+
+import * as types from '@/store/types'
+
+@Component
+export default class SessionTimerMixin extends Vue {
+  startNewNrTimer (): Promise<any> {
+    return this.startTimer({
+      id: NR_COMPLETION_TIMER_NAME,
+      expirationFn: () => {
+        this.$store.dispatch(types.SHOW_NR_SESSION_EXPIRY_MODAL)
+          .then(() => {})
```

Comment:
This is a bit odd. Should this code be waiting for the dispatch call to finish? Or we don't care what happens (ie, empty then function)?

I would expect to see something like this (from onTimerModelSessionExtended):
```
expirationFn: async () => {
   await this.$store.dispatch(types.SHOW_NR_SESSION_EXPIRY_MODAL)
}```

Note that the arrow function is async, and it awaits the dispatch call.

Reply:
We don't care what happens. I am aware of how to use async and ES fat arrow functions as well...
Awaiting the display of that NR session expiry modal blocks I/O and slows things down and I explicitly don't want that to occur. When the session expiry modal displays, a new timer starts for 45 seconds (or whatever that window is set to).

Reply:
I know you are aware of async and arrow functions. I have seen lots of your code use them! 😃 Sorry, I thought this might be an oversight or incomplete code.

Why bother having a "then" block? I think you can just leave it off. Either way, a comment would explain this and prevent a future developer from scratching their head.

Reply:
The then block prevents a linting issue with the current settings that does not allow an unawaited or then'ed call on an async function. I don't want to await those calls, so I just added the empty then block to get around the warning... I left notes in the code in most of the places I did that: 
```
// Don't await this request, that way there's no lag, fire it off async and don't block I/O	      // Don't await this request, that way there's no lag, fire it off async and don't block I/O
      // The empty then clause just prevents a linting issue that warns when you don't	      // The empty then clause just prevents a linting issue that warns when you don't
      // await an async function which is not an issue, since we don't want to block I/O
```

Reply:
OK, cool.